exec/java-exec/src/main/java/org/apache/drill/exec/ops/QueryContext.java

```diff
@@ -91,7 +91,12 @@ public QueryContext(final UserSession session, final DrillbitContext drillbitCon
     executionControls = new ExecutionControls(queryOptions, drillbitContext.getEndpoint());
     plannerSettings = new PlannerSettings(queryOptions, getFunctionRegistry());
     plannerSettings.setNumEndPoints(drillbitContext.getBits().size());
-    table = new DrillOperatorTable(getFunctionRegistry(), drillbitContext.getOptionManager());
+
+    if (getOption(ExecConstants.USE_DYNAMIC_UDFS).bool_val) {
+      table = new DrillOperatorTable(getFunctionRegistry(), drillbitContext.getOptionManager());
```

Comment:
Avoid calling methods in this class in ctor. Use

`this.table = new DrillOperatorTable(drillbitContext.getFunctionImplementationRegistry(), drillbitContext.getOptionManager());`